File: Templates/Pyton/bot.py
import math
from typing import List

from gameElements import FleetCommand, GameState, Position


def GameUpdate(state : GameState) -> List[FleetCommand]:
    me = 1
    output = []

    myPlanets = list(filter(lambda x: x.OwnerId == me, state.Planets))
    enemies = list(filter(lambda x: x.OwnerId != me, state.Planets))

    if myPlanets and enemies:
        for planet in myPlanets:
            enemy = next(iter(sorted(enemies, key=lambda x:Distance(planet.Position, x.Position))))
            output.append(FleetCommand(planet.Id, enemy.Id, planet.NumberOfShips - 1))

    return output

def Distance(p1: Position, p2: Position):
    return math.sqrt((p1.X - p2.X) * (p1.X - p2.X) + (p1.Y - p2.Y) * (p1.Y - p2.Y))

File: Templates/CSharp/Bot.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace SpaceBattlez.CSharp
{
	public class Bot : IBot
	{
		public List<GameElements.FleetCommand> GameUpdate(GameElements.GameState state)
		{
			const int me = 1;
			var output = new List<GameElements.FleetCommand>();

			foreach (var myPlanet in state.Planets.Where(x => x.OwnerId == me))
			{
				var enemy = state.Planets.Where(x => x.OwnerId != me).OrderBy(x => Distance(myPlanet.Position, x.Position)).FirstOrDefault();

				if (enemy != null)
				{
					output.Add(new GameElements.FleetCommand(myPlanet.Id, enemy.Id, myPlanet.NumberOfShips - 1));
				}
			}

			return output;
		}

		private double Distance(GameElements.Position p1, GameElements.Position p2)
		{
			return Math.Sqrt((p1.X - p2.X) * (p1.X - p2.X) + (p1.Y - p2.Y) * (p1.Y - p2.Y));
		}
	}
}